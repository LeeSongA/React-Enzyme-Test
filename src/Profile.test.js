import React from 'react';
import { mount } from 'enzyme';
import Profile from './Profile';

describe('<Profile />', () => {
    it('matches snapshot', () => {
        const wrapper = mount(<Profile username="velopert" name="Song" />);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders username add name', () => {
        const wrapper = mount(<Profile username="velopert" name="Song" />);
        expect(wrapper.props().username).toBe('velopert');
        expect(wrapper.props().name).toBe('Song');

        const boldElement = wrapper.find('b');
        expect(boldElement.contains('velopert')).toBe(true);
        const spanElement = wrapper.find('span');
        expect(spanElement.text()).toBe('(Song)');
    });
});