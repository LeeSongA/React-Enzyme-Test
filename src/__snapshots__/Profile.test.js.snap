// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<Profile
  name="Song"
  username="velopert"
>
  <div>
    <b>
      velopert
    </b>
    Â 
    <span>
      (
      Song
      )
    </span>
  </div>
</Profile>
`;
